### PR TITLE
[JENKINS-35206] - Make the InstallState object deserialization robust against corrupted files

### PR DESCRIPTION
**Pull request by oleg-nenashev**
It's and alternative fix for [JENKINS-35206](https://issues.jenkins-ci.org/browse/JENKINS-35206). This fix may be merged together with #2385, because that PR will be still useful for garbage statuses coming from API calls.
- [x] - Fallback to UNKNOWN install state in the case of corrupted XML
- [x] - Fallback to the UNKNOWN state if the status does not exist anymore (e.g. plugin has been disabled)
- [x] - Unit tests

@reviewbybees @kzantow

### CODE REVIEW

File: test/src/test/java/jenkins/install/InstallStateTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Oleg Nenashev.
```
Comment (oleg-nenashev) [inline test/src/test/java/jenkins/install/InstallStateTest.java]: Oops, automatic code generation

File: core/src/main/java/jenkins/install/InstallState.java [hunk 2]
```diff
@@ -156,6 +160,23 @@ public InstallState(@Nonnull String name, boolean isSetupComplete) {
...
+    public Object readResolve() {
+        // If we get invalid state from the configuration, fallback to unknown
+        if (StringUtils.isBlank(name)) {
+            LOGGER.log(Level.WARNING, "Read install state with blank name: '{0}'. It will be ignored", name);
+            return UNKNOWN;
+        }
+        
+        InstallState state = InstallState.valueOf(name);
+        if (state == null) {
+            LOGGER.log(Level.WARNING, "Cannot locate an extension point for the state '{0}'. It will be ignored", name);
```
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/install/InstallState.java]: Maybe this routine is a subject for another bug

**Comment by kzantow** [comment]
:bee: LGTM

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by oleg-nenashev** [comment]
@jenkinsci/code-reviewers I would be grateful for additional reviews

**Pull request merged**